source/feature/openvswitch/openvswitch-spec.html.md

```diff
@@ -0,0 +1,183 @@
+| Interface        |   X    |   X*  | Interfaces may be created by OVS on the kernel and attached to OVS ports. |
+| Port/NIC         |   X    |   X   |               |
+| Bond             |   X    |   X   | kernel mod    |
+| Bridge           |   X    |   X   |               |
+| Tagging & Tunneling | X   |   X   |               |
+| QoS              |   X    |   X   |               |
+| IP routing       |   X    | w/Controller |        |
+| TCP/UDP stack    |   X    |   -   |               |
+
+### Linux native networking model
+![Linux native networking model](linux_native_networking_model.jpg)
+
+### OVS networking model 
+![OVS networking model](ovs_networking_model.jpg)
+
+
+## Specifications
+
+### Solution Diagram
+![Solution Diagram](solution_diagram.jpg)
+
+### NetSwitch
+A generic switch interface, relaying setup or report requests to the relevant switch implementation (currently two, legacy linux networking and ovs).
+
+This aproach is prefered over having an additional configurator type under the existing switch scheme:
+
+- Adding OVS as a configurator may require changes in the upper common levels, which may introduce instability in existing functionality.
+The proposed abstraction level is aimed to leave existing logic isolated from OVS (and vice versa), reducing potential regression and instability.
+- Different switches or domains should be decoupled, each having their own underling implementation options allowing full flexability on how they are implemented.
+- The modeling of each switch is different, having it seperated reduces complex logic.
+
+Switch implementations are registered per availability, with an identification and a callback. 
+setupNetworks requests are expected to arrive from Engine with a switch type ID, based on which the request is forwarded to the switch implementation.
+
+
+### OVS Configurators
+There are at least two ways to configure OVS: Using the command line utilities (i.e ovs-vsctl) and OVSDB protocol python binding. The latter is considered faster.
+To allow future optimizations, an abstraction is defined with the command line as its implementation. This adopts similar work done on openstack ovsdb agent.
+API Ref: https://review.openstack.org/#/c/143709/20/neutron/agent/ovsdb.py
+
+### IP Interface
+Host IP settings are defined on the kernel stack, independent of the network switch type.
+Current Linux-native networking implementation mixes between the two. For OVS, we would like to put things in better order: the IP setup actions are to be organized under the same “roof”, implemented using configurators.
+
+### Persistence
+OVS persistency is embedded in the switch.
+IP settings are not included and may be implemented using the current persistence scheme.
+- Note: An alternative may be to use a minimal persistent management settings and zero persistency for all the rest. Nevertheless these changes are out of scope and are to be handled under a zero persistency feature. 
+
+### Rollback
```

Comment:
IP config is done outside of OVS transaction, and only after it we would notice that we broke connectivity. It would be great if OVS had a two-phase commitment, so we could configure it first, but commit it only when we have checked Engine connectivity.